core/modules.rs

```diff
@@ -355,12 +364,25 @@ impl Deps {
 
           Self::helper(seen, new_prefix, new_is_last, modules, dep_name)
         }).collect();
-      Deps {
+
+      // If any of the children are missing, return None.
```

Comment:
Some notes on this section (up to line 378):

* I get the clippy warning: `this block may be rewritten with the `?` operator`. Not sure what it has in mind exactly though.
* The whole thing can be written shorter and more efficiently by relying on the fact that the `collect()` method can turn an iterator over `Option<T>` into an `Option<Vec<T>` (this also works for `Result` and for different types of collections). Replace it with this: `let deps = maybe_deps.into_iter().collect::<Option<_>>()?;`